File: web-ui/src/main/java/io/pivotal/demo/domain/WordCount.java
package io.pivotal.demo.domain;

//import org.springframework.data.annotation.Id;
//import org.springframework.data.gemfire.mapping.annotation.Region;

// Word count domain object to represent aggregate word count
// create region --name=Words_Count --type=PARTITION --enable-statistics=true --entry-time-to-live-expiration=60 --entry-time-to-live-expiration-action=DESTROY
// alter runtime --enable-statistics=true --statistic-archive-file=myStatisticsArchiveFile.gfs 
//@Region(name="Words_Count")
public class WordCount {

	//@Id
	private String word;
	private long wordCount;

	@Override
	public String toString() {
		return "WordCount{" +
				"word='" + word + '\'' +
				", count=" + wordCount +
				'}';
	}

	public WordCount() {
	}

	public WordCount(String word, long count) {
		this.word = word;
		this.wordCount = count;
	}

	public String getWord() {
		return word;
	}

	public void setWord(String word) {
		this.word = word;
	}

	public long getWordCount() {
		return wordCount;
	}

	public void setWordCount(long count) {
		this.wordCount = count;
	}
}

File: web-ui/src/main/kotlin/io/pivotal/demo/domain/WordCount.kt
package io.pivotal.demo.domain

import org.springframework.data.annotation.Id
import org.springframework.data.gemfire.mapping.annotation.Region

@Region(name = "Words_Count")
class WordCount {

    @Id
    var word: String? = null
    var wordCount: Long = 0

    override fun toString(): String {
        return "WordCount{" +
                "word='" + word + '\''.toString() +
                ", count=" + wordCount +
                '}'.toString()
    }

    constructor() {}

    constructor(word: String, count: Long) {
        this.word = word
        this.wordCount = count
    }

    operator fun compareTo(other: WordCount): Int {
        val last = this.wordCount.compareTo(other.wordCount)
        return if (last == 0) this.wordCount.compareTo(other.wordCount) else last
    }
}
